kubernetes/linux/main.sh

```diff
@@ -259,16 +259,15 @@ else
 fi
 
 if [ $IS_GET_PODS_API_SUCCESS == 1 ]; then
-      podsResponse=$(cat podsResponseFile)
-      ITEMS_COUNT=$(echo $podsResponse | jq '.items | length')
+      ITEMS_COUNT=$(cat podsResponseFile | jq '.items | length')
       echo "found items count: $ITEMS_COUNT"
       if [ $ITEMS_COUNT -gt 0 ]; then
-            # exclude the pods which doesnt have containerId. could happen if the container fails to start because of bad image and tag etc..
-            podsWithValidContainerId=$(echo $podsResponse | jq -r '[.items[] | select( .status.containerStatuses != null and .status.containerStatuses[].containerID != null and  .status.containerStatuses[].containerID != "")]')
-            ITEMS_COUNT_WITH_CONTAINER_ID=$(echo $podsWithValidContainerId | jq '. | length')
-            if [ $ITEMS_COUNT_WITH_CONTAINER_ID -gt 0 ]; then
-                  containerRuntime=$(echo $podsWithValidContainerId | jq -r '.[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
-                  nodeName=$(echo $podsWithValidContainerId | jq -r '.[0].spec.nodeName')
+            # select omsagent pods
+            omsagentPods=$(cat podsResponseFile | jq -r '[.items[] | select( any(.metadata.name; startswith("omsagent-")))]')
+            ITEMS_COUNT_WITH_OMS_AGENT=$(echo $omsagentPods | jq '. | length')
+            if [ $ITEMS_COUNT_WITH_OMS_AGENT -gt 0 ]; then
+                  containerRuntime=$(echo $omsagentPods | jq -r '.[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
+                  nodeName=$(echo $omsagentPods | jq -r '.[0].spec.nodeName')
```

Comment:
Why do we need this tool, rather use python for json parsing , which is reliable & tested ?

Reply:
> Why do we need this tool, rather use python for json parsing , which is reliable & tested ?
> Let me explore Python JSON parser and see if that is better and works for all the scenarios particularly with larger payloads.  Jq is popular tool and well tested as well.